backend/controllers/retentionReasons.js

```diff
@@ -42,7 +72,23 @@ const retentionReasonsFactory = (elite2Api, dataComplianceApi, logError) => {
 
   const index = async (req, res) => renderTemplate(req, res)
 
-  return { index }
+  const post = async (req, res) => {
+    try {
+      const { offenderNo } = req.params
+      const { reasons, version } = qs.parse(req.body)
```

Comment:
Using qs here to very simply group reason codes and details together into an array.  Without this, I'd need some custom JS to parse the form inputs in the request body, handling things like singular checkboxes generating strings, not arrays of strings...